test/src/analyzers/lint_analyzer/rules/rules_list/avoid_useless_type_manipulation/avoid_useless_type_manipulations_test.dart

```diff
@@ -0,0 +1,200 @@
+import 'package:dart_code_metrics/src/analyzers/lint_analyzer/models/severity.dart';
+import 'package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/avoid_useless_type_manipulations/avoid_useless_type_manipulations.dart';
+import 'package:test/test.dart';
+
+import '../../../../../helpers/rule_test_helper.dart';
+
+const _path = 'avoid_useless_type_manipulation/examples';
+const _classExampleWithIs = '$_path/example_with_is.dart';
+const _classExampleWithAs = '$_path/example_with_as.dart';
+const _classExampleCases = '$_path/example_cases.dart';
+
+void main() {
+  group('AvoidUselessTypeManipulations', () {
+    test('initialization', () async {
+      final unit = await RuleTestHelper.resolveFromFile(_classExampleWithIs);
+      final issues = AvoidUselessTypeManipulations().check(unit);
+
+      RuleTestHelper.verifyInitialization(
+        issues: issues,
+        ruleId: 'avoid-useless-type-manipulations',
+        severity: Severity.style,
+      );
+    });
+
+    test('reports about found all issues in example_with_is.dart', () async {
+      final unit = await RuleTestHelper.resolveFromFile(_classExampleWithIs);
+      final issues = AvoidUselessTypeManipulations().check(unit);
+
+      RuleTestHelper.verifyIssues(
+        issues: issues,
+        startOffsets: [
+          120,
+          228,
+          539,
+          584,
+          630,
+          672,
+          718,
+          1020,
+          1053,
+        ],
+        startLines: [
+          6,
+          8,
+          21,
+          22,
+          23,
+          24,
+          25,
+          38,
+          39,
+        ],
+        startColumns: [
+          20,
+          21,
+          20,
+          20,
+          20,
+          20,
+          20,
+          20,
+          5,
+        ],
+        endOffsets: [
+          143,
+          252,
+          555,
+          601,
+          643,
+          689,
+          731,
+          1039,
+          1076,
+        ],
+        locationTexts: [
+          'regularString is String',
+          'regularString is String?',
+          'animal is Animal',
+          'cat is HomeAnimal',
+          'cat is Animal',
+          'dog is HomeAnimal',
+          'dog is Animal',
+          'myList is List<int>',
+          'myList.whereType<int>()',
+        ],
+        messages: [
+          'Avoid useless is operation regularString is String.',
```

Comment:
The more I look at this error message, the more I think that the example at the end should be omitted, since it's pretty confusing without quotes. I think `Avoid redundant "is" assertion` would be the best choice here.